File: flutter/openral_flutter/lib/model/specific_property.dart
class SpecificProperty {
  final String key;
  final dynamic value;
  final String? unit;

  SpecificProperty({
    required this.key,
    required this.value,
    this.unit,
  });

  Map<String, dynamic> toMap() {
    return {
      'key': key,
      'value': value,
      'unit': unit,
    };
  }
}

File: python/openral_py/model/specific_property.py
from typing import Any, Dict, Optional


class SpecificProperty:
    def __init__(self, key: str, value: Any, unit: Optional[str] = None):
        self.key = key
        self.value = value
        self.unit = unit

    def to_map(self):
        return {
            'key': self.key,
            'value': self.value,
            'unit': self.unit,
        }
    
    @staticmethod
    def from_map(property_map: Dict[str, Any]) -> 'SpecificProperty':
        key = property_map['key']
        value = property_map['value']

        unit = None
        if 'unit' in property_map:
            unit = property_map['unit']

        return SpecificProperty(key, value, unit)
